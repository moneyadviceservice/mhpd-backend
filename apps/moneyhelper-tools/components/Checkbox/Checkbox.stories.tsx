import { StoryFn } from '@storybook/react';
import { Checkbox, CheckboxProps } from './Checkbox';

const StoryProps = {
  title: 'Components/TOOLS/CheckboxButton',
  component: Checkbox,
};

const Template: StoryFn<CheckboxProps> = (args) => <Checkbox {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Checkbox',
  value: '',
  id: 'id-1',
  name: 'field-name',
};

export default StoryProps;
