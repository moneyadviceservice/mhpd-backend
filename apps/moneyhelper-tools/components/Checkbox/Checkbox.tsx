import { DetailedHTMLProps, InputHTMLAttributes } from 'react';
import { twMerge } from 'tailwind-merge';
import Tick from '../../public/icons/tick.svg';

export type CheckboxProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
>;

export const Checkbox = ({
  children,
  style,
  className,
  ...rest
}: CheckboxProps) => (
  <label style={style} className={twMerge('flex', className)}>
    <input type="checkbox" className="peer sr-only tool-cbox" {...rest} />
    <div className="flex justify-center items-center flex-none w-10 h-10  p-1  bg-white text-white peer-checked:text-pink-600  border border-gray-400 rounded peer-focus:border-2 peer-focus:border-purple-700  peer-focus:ring-4 peer-focus:ring-yellow-300 peer-disabled:border-slate-400 peer-checked:disabled:text-gray-500">
      <Tick className="w-6 h-5" role="presentation" />
    </div>
    <p className="ml-4 self-center">{children}</p>
  </label>
);
