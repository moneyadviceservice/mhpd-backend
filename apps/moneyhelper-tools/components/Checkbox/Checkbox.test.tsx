import { render } from '@testing-library/react';
import { Checkbox } from './Checkbox';

describe('Checkbox component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Checkbox id="test-id" name="test-name" value="">
        Lorem Ipsum
      </Checkbox>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
