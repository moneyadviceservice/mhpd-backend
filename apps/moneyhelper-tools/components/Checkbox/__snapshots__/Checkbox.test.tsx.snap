// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders correctly 1`] = `
<label
  class="flex"
>
  <input
    class="peer sr-only tool-cbox"
    id="test-id"
    name="test-name"
    type="checkbox"
    value=""
  />
  <div
    class="flex justify-center items-center flex-none w-10 h-10  p-1  bg-white text-white peer-checked:text-pink-600  border border-gray-400 rounded peer-focus:border-2 peer-focus:border-purple-700  peer-focus:ring-4 peer-focus:ring-yellow-300 peer-disabled:border-slate-400 peer-checked:disabled:text-gray-500"
  >
    <svgr-mock
      classname="w-6 h-5"
      role="presentation"
    />
  </div>
  <p
    class="ml-4 self-center"
  >
    Lorem Ipsum
  </p>
</label>
`;
